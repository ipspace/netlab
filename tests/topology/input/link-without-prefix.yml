# Test links without prefix (no IP config)
#
defaults:
  device: eos

nodes:
- r1
- r2
- r3

links:

# P2P link
- r1-r2

# LAN link
- [ r1, r2, r3 ]

# P2P link
- r1:
  r2:
  prefix: false

# LAN link
- r1:
  r2:
  r3:
  prefix: false

