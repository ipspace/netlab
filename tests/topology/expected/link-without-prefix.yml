input:
- topology/input/link-without-prefix.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.1/30
    node: r1
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.2/30
    node: r2
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- bridge: input_2
  interfaces:
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.0.1/24
    node: r1
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.0.2/24
    node: r2
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 172.16.0.3/24
    node: r3
  linkindex: 2
  node_count: 3
  prefix:
    ipv4: 172.16.0.0/24
  type: lan
- interfaces:
  - ifindex: 3
    ifname: Ethernet3
    node: r1
  - ifindex: 3
    ifname: Ethernet3
    node: r2
  linkindex: 3
  node_count: 2
  prefix: false
  type: p2p
- bridge: input_4
  interfaces:
  - ifindex: 4
    ifname: Ethernet4
    node: r1
  - ifindex: 4
    ifname: Ethernet4
    node: r2
  - ifindex: 2
    ifname: Ethernet2
    node: r3
  linkindex: 4
  node_count: 3
  prefix: false
  type: lan
name: input
nodes:
  r1:
    af:
      ipv4: true
    box: arista/veos
    device: eos
    id: 1
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: r1 -> r2
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.2/30
        node: r2
      type: p2p
    - bridge: input_2
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.0.1/24
      linkindex: 2
      name: r1 -> [r2,r3]
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.0.2/24
        node: r2
      - ifname: Ethernet1
        ipv4: 172.16.0.3/24
        node: r3
      type: lan
    - ifindex: 3
      ifname: Ethernet3
      linkindex: 3
      name: r1 -> r2
      neighbors:
      - ifname: Ethernet3
        node: r2
      type: p2p
    - bridge: input_4
      ifindex: 4
      ifname: Ethernet4
      linkindex: 4
      name: r1 -> [r2,r3]
      neighbors:
      - ifname: Ethernet4
        node: r2
      - ifname: Ethernet2
        node: r3
      type: lan
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    name: r1
  r2:
    af:
      ipv4: true
    box: arista/veos
    device: eos
    id: 2
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: r2 -> r1
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.1/30
        node: r1
      type: p2p
    - bridge: input_2
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.0.2/24
      linkindex: 2
      name: r2 -> [r1,r3]
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: Ethernet1
        ipv4: 172.16.0.3/24
        node: r3
      type: lan
    - ifindex: 3
      ifname: Ethernet3
      linkindex: 3
      name: r2 -> r1
      neighbors:
      - ifname: Ethernet3
        node: r1
      type: p2p
    - bridge: input_4
      ifindex: 4
      ifname: Ethernet4
      linkindex: 4
      name: r2 -> [r1,r3]
      neighbors:
      - ifname: Ethernet4
        node: r1
      - ifname: Ethernet2
        node: r3
      type: lan
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    name: r2
  r3:
    af:
      ipv4: true
    box: arista/veos
    device: eos
    id: 3
    interfaces:
    - bridge: input_2
      ifindex: 1
      ifname: Ethernet1
      ipv4: 172.16.0.3/24
      linkindex: 2
      name: r3 -> [r1,r2]
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.0.1/24
        node: r1
      - ifname: Ethernet2
        ipv4: 172.16.0.2/24
        node: r2
      type: lan
    - bridge: input_4
      ifindex: 2
      ifname: Ethernet2
      linkindex: 4
      name: r3 -> [r1,r2]
      neighbors:
      - ifname: Ethernet4
        node: r1
      - ifname: Ethernet4
        node: r2
      type: lan
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    name: r3
provider: libvirt
